components/linkedin/linkedin.app.mjs

```diff
@@ -1,11 +1,158 @@
+import { axios } from "@pipedream/platform";
+import constants from "./common/constants.mjs";
+
 export default {
   type: "app",
   app: "linkedin",
-  propDefinitions: {},
+  propDefinitions: {
+    text: {
+      type: "string",
+      label: "Text",
+      description: "Text to be posted on LinkedIn timeline",
+    },
+    organizationId: {
+      type: "string",
+      label: "Organization Id",
+      description: "ID of the organization that will author the post",
+    },
+    type: {
+      type: "string",
+      label: "Type",
+      description: "Type of the post",
+      options: [
+        {
+          label: "Text/URL",
+          value: "NONE",
+        },
+        {
+          label: "Article",
+          value: "ARTICLE",
+        },
+      ],
+      optional: false,
+    },
+    visibility: {
+      type: "string",
+      label: "Visibility",
+      description: "Visibility restrictions on content",
+      options: [
+        {
+          label: "Connections",
+          value: "CONNECTIONS",
+        },
+        {
+          label: "Public",
+          value: "PUBLIC",
+        },
+        {
+          label: "Logged in",
+          value: "LOGGED_IN",
+        },
+      ],
+      optional: false,
+    },
+    postId: {
+      type: "string",
+      label: "Post ID",
+      description: "Id of the post that will be deleted",
+      optional: false,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getHeaders() {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "X-Restli-Protocol-Version": "2.0.0",
+      };
+    },
+    async _makeRequest(customConfig) {
```

Comment:
Optionally, the parameter here can be given like;

```
...
async _makeRequest({
    $,
    url,
    path,
    ...otherConfig
} = {}) {
    const BASE_URL = constants.BASE_URL;
...
```
and they can be used like;
```
...
const config = {
    url: url || `${BASE_URL}${constants.VERSION_PATH}${path}`,
    headers: this._getHeaders(),
    ...otherConfig
};
...
```
I think this one might be more readable.